chutney/server/src/main/java/com/chutneytesting/campaign/infra/CampaignExecutionDBRepository.java

```diff
@@ -164,23 +161,23 @@ public Long generateCampaignExecutionId(Long campaignId, String environment) {
         campaignExecutionJpaRepository.save(newExecution);
         return newExecution.id();
     }
-    private CampaignExecution toDomain(CampaignExecutionEntity campaignExecution) {
-        return toDomainWithCampaign(
-            campaignJpaRepository.findById(campaignExecution.campaignId()).orElseThrow(() -> new CampaignExecutionNotFoundException(campaignExecution.campaignId(), campaignExecution.id())),
-            campaignExecution);
-    }
 
-    private CampaignExecution toDomainWithCampaign(CampaignEntity campaign, CampaignExecutionEntity campaignExecutionEntity) {
-        return campaignExecutionEntity.toDomain(campaign, isCampaignExecutionRunning(campaignExecutionEntity), this::title);
+    private CampaignExecution toDomain(CampaignExecutionEntity campaignExecution, Supplier<? extends RuntimeException> campaignExceptionSupplier) {
+        CampaignEntity campaign = campaignJpaRepository.findById(campaignExecution.campaignId()).orElseThrow(campaignExceptionSupplier);
```

Comment:
dont understand why we have a different error message if we cannot find the campaign